src/windows.rs

```diff
@@ -0,0 +1,343 @@
+/// very often. Also, there's limited (though huge) amount of distinct IDs (they are `u64`).
+///
+/// # Examples
+///
+/// ```rust
+/// extern crate signal_hook;
+///
+/// use std::io::Error;
+/// use std::process;
+///
+/// fn main() -> Result<(), Error> {
+///     let signal = unsafe { signal_hook::register(signal_hook::SIGTERM, || process::abort()) }?;
+///     // Stuff here...
+///     signal_hook::unregister(signal); // Not really necessary.
+///     Ok(())
+/// }
+/// ```
+pub unsafe fn register<F>(signal: c_int, action: F) -> Result<SigId, Error>
+where
+    F: Fn() + Sync + Send + 'static,
+{
+    assert!(
+        !FORBIDDEN.contains(&signal),
+        "Attempted to register forbidden signal {}",
+        signal,
+    );
+    let data = GlobalData::ensure();
+    let mut state = data.state.lock().unwrap();
+    let action_id = ActionId(state.action_id);
+    state.action_id += 1;
+
+    let sig_id = SigId {
+        signal,
+        action: action_id,
+    };
+
+    let all_signals = state
+        .all_signals
+        .entry(signal)
+        .or_insert_with(BTreeMap::new);
+    all_signals.insert(action_id, Box::new(action));
+
+    Ok(sig_id)
+}
+
+/// Removes a previously installed action.
+///
+/// This function does nothing if the action was already removed. It returns true if it was removed
+/// and false if the action wasn't found.
+///
+/// It can unregister all the actions installed by [`register`](fn.register.html) as well as the
+/// ones from helper submodules.
+///
+/// # Warning
+///
+/// This does *not* currently return the default/previous signal handler if the last action for a
+/// signal was just unregistered. That means that if you replaced for example `SIGTERM` and then
+/// removed the action, the program will effectively ignore `SIGTERM` signals from now on, not
+/// terminate on them as is the default action. This is OK if you remove it as part of a shutdown,
+/// but it is not recommended to remove termination actions during the normal runtime of
+/// application (unless the desired effect is to create something that can be terminated only by
+/// SIGKILL).
+pub fn unregister(id: SigId) -> bool {
+    let data = GlobalData::ensure();
+    let mut state = data.state.lock().unwrap();
+    let mut replace = false;
+    if let Some(actions) = state.all_signals.get_mut(&id.signal) {
+        replace = actions.remove(&id.action).is_some();
+    }
+    replace
+}
+
+// Emulates kill(2) for testing `signal-hook` in Windows.
+#[doc(hidden)]
+pub unsafe fn __emulate_kill(pid: c_int, signal: c_int) -> c_int {
+    let self_pid = ::libc::getpid();
+    if pid != self_pid && pid != -self_pid && pid != 0 && pid != -1 {
+        return 0;
+    }
+
+    let data = GlobalData::ensure();
+    let state = data.state.lock().unwrap();
+    if let Some(all_signals) = state.all_signals.get(&signal) {
+        for action in all_signals.values() {
+            action();
+        }
+    }
+    0
+}
```

Comment:
I understand you want to use this inside the tests. But AFAIK this actually doesn't test anything (or, only part), because it doesn't go through the signal/console control.

Is there no way to send the real thing in Windows?

Reply:
As mentioned, I've missed it and it actually exists.